provisioning/hare_setup.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import argparse
+import asyncio
+import json
+import logging
+import shutil
+import subprocess
+import os
+import yaml
+import socket
+
+from cortx.utils.product_features import unsupported_features
+
+def execute_command_and_get_output(cmd):
+        process = subprocess.Popen(cmd, stdin=subprocess.PIPE,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        shell=True)
+        stdout, stderr = process.communicate()
+        output_var = str(stdout, 'utf-8')
+        return output_var
+
+
+def setup_logging():
+    logging.basicConfig(level=logging.INFO, format='%(message)s')
+
+
+def get_data_from_provisioner_cli(method, output_format='json') -> str:
+    try:
+        process = subprocess.run(['provisioner', method, f'--out={output_format}'],
```

Comment:
Why don't you add a helper function that invokes the given command properly (like [`_execute()`](https://github.com/knekrasov/cortx-ha/blob/main/ha/pcswrap/pcswrap/internal/connector.py#L72) in my previous example from pcswrap).

That will make the code shorter and will hide the ugly details of `subprocess.run()`. And Codacy findings can be addressed there, within a single place :wink:

Reply:
@knekrasov This is existing code showing here because of file rename. Can I have handle it in separate task as provisioner setup will be required to test the changes

Reply:
Well, I think that provisioner setup just to ensure that the code is not broken is an overkill, but OK, let's  defer this refactoring. Please ignore the comment and concentrate on your changes.